server/src/buffer.c

```diff
@@ -20,16 +29,22 @@ buffer_t *create_buffer(size_t capacity)
 
 void buffer_write(buffer_t *buffer, char *data)
 {
-    int len = strlen(data);
+    size_t data_size = strlen(data);
 
-    if (!can_write(buffer, len))
+    if (!can_write(buffer, data_size))
+        return;
+    if (data[0] == '\0' || data[0] == '\n' || data[0] == '\r')
         return;
-    for (int i = 0; i < len; i++) {
+    for (size_t i = 0; i < data_size; i++) {
         if (buffer->write_index == buffer->capacity)
             buffer->write_index = 0;
+        if (data[i] == 13)
```

Comment:
Where does 13 comes from ?

Reply:
It's the ascii code the '\r' because we considered '\r' as a useless char for our server